### ISSUE TITLE
Open emails/bundles inline (instead of navigating to a new page)

### ISSUE THREAD

**Issue by digitalneoplasm** [boukestam/inbox-in-gmail#79]
In Inbox, when you clicked on a mail item it didn't take you to a new page as gmail does. Instead, it opened in-line with the rest of your inbox. I imagine this is a massive undertaking, and perhaps not even possible given with is available when doing this as a Chrome extension, but I figured I'd put it out there :). 

![Pasted_Image_5_8_19__2_03_PM](https://user-images.githubusercontent.com/1452510/57397334-71396580-719a-11e9-8a9d-92ac4172f6af.png)

It also worked this way for bundles - they would be visible in-line as well: 

![image](https://user-images.githubusercontent.com/1452510/57397542-f58be880-719a-11e9-8199-22db33a58902.png)

Either or both would be great!

**Comment by russelldc** [comment]
@digitalneoplasm This has definitely been considered, but like you said:

>this is a massive undertaking

I don't think it's impossible at all, though. I have an issue up for bundle enhancements (#70), including improving the UX for navigation. It seems like spending time making all those changes would be wasted effort if this is possible instead.

If no one else gets to this first, I think I'll try something out. I'd go with inline bundles first, because I really don't like navigating back and forth to those search pages (although it's a whole lot better compared to not having them at all).

If it works out well for bundles, it shouldn't be too hard to extend that work to emails as well.

For inline bundles, I think the hardest part would be repositioning the emails from other date sections. The rest of the work is just:
- Remove the current link to the search page
- Add a click handler to closed bundles that:
  - Closes any other bundles if they're open
  - Apply a class to the clicked bundle to animate into inline bundle
  - Apply a class to the bundled emails (with the same label/category) that cause them to appear
- Add a click handler to open bundles that:
  - Closes the current bundle (by removing the classes from the bundle and bundled emails)

**Comment by Muffo** [comment]
This would be awesome!

**Comment by davelavoie** [comment]
Hi there,

I've tested the following script, and while it doesn't open emails inline, it does open the bundles inline:
https://github.com/TimeBomb/bundles-gmail

If I disable the bundles in the Inbox Reborn plugin settings, and enable the Bundles Gmail script, it almost works out of the box. The only issue is caused by the way emails are grouped by date: when opening a bundle, if one of the email was from another time period (yesterday, last month), it'll be displayed under these sections.

It would be nice if Inbox Reborn includes this script! :-)

**Comment by whitespacecode** [comment]
> Hi there,
> 
> I've tested the following script, and while it doesn't open emails inline, it does open the bundles inline:
> https://github.com/TimeBomb/bundles-gmail
> 

Thanks for this.
I just came across these git pages!! I will try to merge the bundles scripts in the inbox in gmail.

**Comment by russelldc** [comment]
There's a lot about that script/implementation that I don't like, but I guess that just means we need to improve it!

**Comment by whitespacecode** [comment]
> There's a lot about that script/implementation that I don't like, but I guess that just means we need to improve it!

Yes i see lots of lines you don't really need.. But the core functionality works and that is what matters :)
 When i find a bit of time i'm going to look into it..

**Comment by davelavoie** [comment]
I suppose it'll probably need some testing before it gets eventually merged to the master branch. I will be happy to help you during the testing phase!
